File: step1.py
import time

def task(name):
    print(f"Starting task {name}")
    time.sleep(2)
    print(f"Task {name} completed")

if __name__ == "__main__":
    start = time.time()

    task("A")
    task("B")

    end = time.time()
    print(f"Total time: {end - start:.2f} seconds")

File: step1.go
// multithreading
package main

import (
	"fmt"
	"runtime"
	"sync"
)

func cpuHeavy(name string, wg *sync.WaitGroup) {
	defer wg.Done()
	fmt.Printf("Starting %s\n", name)
	total := 0
	for i := 0; i < 10_000_000; i++ {
		total += i * i
	}
	fmt.Printf("%s complete\n", name)
}

func main() {
	runtime.GOMAXPROCS(runtime.NumCPU()) // Use all available CPU cores
	var wg sync.WaitGroup
	wg.Add(2)

	go cpuHeavy("Goroutine A", &wg)
	go cpuHeavy("Goroutine B", &wg)

	wg.Wait()
}
